basyx.aasservice/basyx.aasservice-backend-inmemory/src/main/java/org/eclipse/digitaltwin/basyx/aasservice/backend/InMemoryAasService.java

```diff
@@ -58,8 +64,28 @@ public AssetAdministrationShell getAAS() {
 	}
 
 	@Override
-	public List<Reference> getSubmodelReferences() {
-		return aas.getSubmodels();
+	public CursorResult<List<Reference>> getSubmodelReferences(PaginationInfo pInfo) {
```

Comment:
Please refactor according to clean code (e.g., no comments, split into subfunctions)